### PR TITLE
github-ci: provide commit id within manual trigger

### PR DESCRIPTION
**Pull request by drakonhg**
Git checkout was added to workflows where packaging is located
to check if workflow_dispatch is working correctly or not

There is three types of testing
  - Short commit hash
  - Long commit hash
  - Without hash

### CODE REVIEW

**Comment by VitaliyaIoffe** [comment]
Seems good!
image

**Comment by VitaliyaIoffe** [comment]
Please, fix the commit message. 

Firstly, you wrote about testing formats. That's good info, but it is not enough. Write about why this change is useful. 
Secondly, your commit title `Add worker_dispatch of commit hash to all existing workflows` has length reater than 72 characters. 72 characters is a recommended length for a title. Also, it's good to start title from 'github-ci: <explanation>'. 
Maybe something like this:
`github-ci: provide commit hash as a manual trigger`
What do you think?

**Comment by VitaliyaIoffe** [comment]
https://github.com/tarantool/tarantool/runs/3320415906?check_suite_focus=true

**Comment by drakonhg** [comment]
> Please, fix the commit message.
> 
> Firstly, you wrote about testing formats. That's good info, but it is not enough. Write about why this change is useful. Write what would be done after you put full hash, small hash, wrong hash and no hash.
> Secondly, your commit title `Add worker_dispatch of commit hash to all existing workflows` has a length greater than 50 characters. 50 characters is a recommended length for a title. Also, it's good to start title from 'github-ci: '.
> Maybe something like this:
> `github-ci: provide commit id within manual trigger`
> What do you think?

Sounds great. Will do

**Review by VitaliyaIoffe — CHANGES_REQUESTED** [review]
Tarantool involves a few submodules. If we checkout input a commit only the tarantool repo and packaging it, it would be inconsistent, cause we haven't updated submodules. 
So, after checkout we should fetch all submodules.

File: .github/workflows/ubuntu_20_04.yml [hunk 1]
```diff
@@ -30,7 +35,7 @@ jobs:
       - uses: actions/checkout@v2.3.4
         with:
           fetch-depth: 0
-          submodules: recursive
```
Comment (VitaliyaIoffe) [inline .github/workflows/ubuntu_20_04.yml:33]: return please

**Review by VitaliyaIoffe — CHANGES_REQUESTED** [review]
Also, this issue is about packaging. Please, left the change only in workflows, where there is packaging step.

File: .github/workflows/source.yml [hunk 2]
```diff
@@ -34,6 +38,9 @@ jobs:
         with:
           fetch-depth: 0
           submodules: recursive
+      - name: Checkout input commit hash
+        run: |
+          git checkout ${{ github.event.inputs.commit }}
```
Comment (VitaliyaIoffe) [inline .github/workflows/source.yml:43]: delete please this step everywhere.

File: .github/workflows/source.yml [hunk 3]
```diff
@@ -5,6 +5,10 @@ on:
   repository_dispatch:
     types: [backend_automation]
   workflow_dispatch:
+    inputs:
+      commit:
+        description: "Commit id or Branch Name"
+        required: false
```
Comment (VitaliyaIoffe) [inline .github/workflows/source.yml:11]: The issue is about packaging, please delete this change from workflows without packaging or make this change same on all workflows.

**Review by VitaliyaIoffe — CHANGES_REQUESTED** [review]

File: .github/workflows/ubuntu_21_04.yml [hunk 4]
```diff
@@ -6,6 +6,10 @@ on:
   repository_dispatch:
     types: [backend_automation]
   workflow_dispatch:
+    inputs:
+      commit:
+        description: "Commit id"
+        required: true
```
Comment (VitaliyaIoffe) [inline .github/workflows/ubuntu_21_04.yml:12]: why is here `true`?

File: .github/workflows/fedora_33.yml [hunk 5]
```diff
@@ -6,6 +6,11 @@ on:
   repository_dispatch:
     types: [backend_automation]
   workflow_dispatch:
+    inputs:
+      commit:
+        description: "Commit id"
+        required: false
+
```
Comment (VitaliyaIoffe) [inline .github/workflows/fedora_33.yml:13]: blank line

**Review by VitaliyaIoffe — CHANGES_REQUESTED** [review]

**Review by VitaliyaIoffe — APPROVED** [review]
LGTM

**Comment by VitaliyaIoffe** [comment]
@igormunkin, do we need to upload packages anywhere?

**Review by VitaliyaIoffe — CHANGES_REQUESTED** [review]
Should upload packages to artifacts

**Comment by Totktonada** [comment]
It is stalled for several months, so unlikely will receive updates. I'll close this pull request. It is linked to the issue, so the effort will not gone. Someone, who will continue the work, can reuse it.

Feel free to reopen if it is a mistake.

**Pull request closed without merging**